#   Taller Pandas 

In [126]:
import pandas as pd
import numpy as np

El documento base fue: Enfermedades derivadas de la ocupación en unidades de trabajo informal en bogotá; obtenido en el portal de datos abiertos de bogotá.

In [127]:
## Utilicé el sep=";" porque es un archivo csv de excel y el encoding="latin" ya que el archivo tiene ñ.
trabajos_informales = pd.read_csv(r"C:\Users\Usuario\Downloads\obs_saludlab_enfermedaduti_.csv", sep=';', encoding='latin1')


In [128]:
## Para verificar y limpiar los duplicados
dupli = trabajos_informales.duplicated () .sum()
print(f"Hay {dupli} duplicados.")

Hay 265 duplicados.


In [129]:
trabajos_informales = trabajos_informales.drop_duplicates (keep='first')
trabajos_informales = trabajos_informales.reset_index(drop=True)
print("Duplicados eliminados.")

Duplicados eliminados.


In [130]:
## Revisar los nombres y cantidad de columnas, cantidad de valores y tipo de valor
trabajos_informales.info ()

<class 'pandas.DataFrame'>
RangeIndex: 23377 entries, 0 to 23376
Data columns (total 14 columns):
 #   Column                                 Non-Null Count  Dtype
---  ------                                 --------------  -----
 0   Año                                    23377 non-null  int64
 1   Localidad de ocurrencia del caso       23377 non-null  str  
 2   Sexo                                   23377 non-null  str  
 3   Edad                                   23377 non-null  int64
 4   Régimen aseguramiento en salud         23377 non-null  str  
 5   Escolaridad                            23376 non-null  str  
 6   Ocupación                              23375 non-null  str  
 7   Tipo UTI                               23377 non-null  str  
 8   Clase UTI                              23376 non-null  str  
 9   Nivel de ingresos                      23377 non-null  str  
 10  Forma de Pago                          23376 non-null  str  
 11  Síntoma                                

In [131]:
## Con el fin de eliminar filas vacías
trabajos_informales.dropna(inplace=True)

In [132]:
trabajos_informales.info ()

<class 'pandas.DataFrame'>
Index: 23370 entries, 0 to 23376
Data columns (total 14 columns):
 #   Column                                 Non-Null Count  Dtype
---  ------                                 --------------  -----
 0   Año                                    23370 non-null  int64
 1   Localidad de ocurrencia del caso       23370 non-null  str  
 2   Sexo                                   23370 non-null  str  
 3   Edad                                   23370 non-null  int64
 4   Régimen aseguramiento en salud         23370 non-null  str  
 5   Escolaridad                            23370 non-null  str  
 6   Ocupación                              23370 non-null  str  
 7   Tipo UTI                               23370 non-null  str  
 8   Clase UTI                              23370 non-null  str  
 9   Nivel de ingresos                      23370 non-null  str  
 10  Forma de Pago                          23370 non-null  str  
 11  Síntoma                                23370

In [151]:
trabajos_informales['Localidad de ocurrencia del caso'].value_counts()

Localidad de ocurrencia del caso
CIUDAD BOLÍVAR        1787
SAN CRISTÓBAL         1645
BARRIOS UNIDOS        1628
RAFAEL URIBE URIBE    1592
SUBA                  1560
USME                  1549
ENGATIVÁ              1534
KENNEDY               1526
BOSA                  1490
USAQUÉN               1230
FONTIBÓN              1185
LOS MÁRTIRES          1180
SANTA FE              1164
ANTONIO NARIÑO        1110
TUNJUELITO            1069
PUENTE ARANDA          663
TEUSAQUILLO            432
CHAPINERO              424
LA CANDELARIA          357
SUMAPAZ                245
Name: count, dtype: int64

In [ ]:
### 1 pregunta. ¿cúal es la localidad con mayor porcentaje de escolaridad primaria incompleta?
## Colocamos el .size().unstack(fill_value=0)  para que corrija los valores y los organize de una forma más facil de leer.
localidad_esc = trabajos_informales.groupby(['Localidad de ocurrencia del caso', 'Escolaridad']).size().unstack(fill_value=0)
rangoprimaria = localidad_esc['Primaria incompleta'].sort_values(ascending=False)

print("LOCALIDADES PRIMARIA INCOMPLETA")
print( rangoprimaria.head(5))

LOCALIDADES PRIMARIA INCOMPLETA
Localidad de ocurrencia del caso
CIUDAD BOLÍVAR    225
USME              187
SAN CRISTÓBAL     178
BOSA              127
TUNJUELITO        125
Name: Primaria incompleta, dtype: int64


In [ ]:
### 2 pregunta. ¿cúal es la ocupación informal con el nivel de ingresos más alto?

# Para revisar los diferentes niveles de ingresos y saber sobre cual trabajar más adelante
trabajos_informales['Nivel de ingresos'].value_counts()

Nivel de ingresos
1 SMMLV                  10428
Entre 1 y 2 SMMLV         7457
Menos de 1 SMMLV          5292
2 y MÁS SMMLV              102
Entre 1 y 2 SMMLV           75
NS / NR                     12
2 y más SMMLV                3
NS/NR                        1
Name: count, dtype: int64

In [ ]:
## Para limpiar los niveles de ingresos repetidos tomados como diferentes por las formas de ingreso de la información
trabajos_informales['Nivel de ingresos'] = trabajos_informales['Nivel de ingresos'].str.strip().str.upper()
trabajos_informales['Nivel de ingresos'] = trabajos_informales['Nivel de ingresos'].replace({'NS/NR': 'NS / NR',
                                                                                             'NS/NRENTRE 1 Y 2 SMMLV ': 'ENTRE 1 Y 2 SMMLV'})



In [ ]:
## Revisar de que los valores fueron corregidos
trabajos_informales['Nivel de ingresos'].value_counts()

Nivel de ingresos
1 SMMLV              10428
ENTRE 1 Y 2 SMMLV     7532
MENOS DE 1 SMMLV      5292
2 Y MÁS SMMLV          105
NS / NR                 13
Name: count, dtype: int64

In [ ]:

ingresos_masaltos = trabajos_informales[trabajos_informales['Nivel de ingresos'] == '2 Y MÁS SMMLV']
trabajomas_ingresos = ingresos_masaltos['Ocupación'].value_counts()

print("OCUPACIONES CON INGRESOS MÁS ALTOS")
print(trabajomas_ingresos.head(4))

OCUPACIONES CON INGRESOS MÁS ALTOS
Ocupación
Peluqueros, especialistas en tratamientos de belleza y afines    24
Vendedores y demostradores de tiendas, almacenes y afines        13
Mecánicos y ajustadores de vehículos de motor                    13
Encargados de control de abastecimiento e inventario             13
Name: count, dtype: int64


In [ ]:
### 3 Pregunta. ¿cuáles son los sintomas más frecuentes en los trabajos informales?
sintomas_frecuentes = trabajos_informales['Síntoma'].value_counts()
print("SÍNTOMAS MÁS FRECUENTES")
print(sintomas_frecuentes.head(5))

SÍNTOMAS MÁS FRECUENTES
Síntoma
Dolor dorsal o lumbar              3753
Dolor miembro(s) inferior(es)      2541
Dolor miembro(s) superior(es)      1826
Dolor dorso lumbar                 1416
Prurito, resequedad y/o fisuras    1293
Name: count, dtype: int64


# Conclusiones:
1. ¿cúal es la localidad con mayores casos de escolaridad primaria incompleta?
Ciudad Bolivar con 225 casos es la localidad reportada con más casos de escolaridad primaria incompleta, en segundo lugar se encuentra usme con 187 casos y en tercer lugar san cristobal con 178 casos. 

2. ¿cúal es la ocupación informal con el nivel de ingresos más alto?
La Proporción acumulada de la ocupación con el nivel de ingresos más alto son Peluqueros, especialistas en tratamientos de belleza y afines es 24 reportando un ingreso de 2 Y MÁS SMMLV, superando a los Vendedores y demostradores de tiendas, almacenes y afines de los cuales se reportaron 11 puntos menos con un ingreso igual de 2 Y MÁS SMMLV. 

3. ¿cuáles son los sintomas más frecuentes en los trabajos informales?
Con 3,753 casos, el dolor dorsal o lumbar es el sintoma más frecuente entre los trabajos informales seguido del dolor de miembros inferiores con 2,541 casos.
